yt_dlp/extractor/fancode.py

```diff
@@ -38,16 +39,54 @@ class FancodeVodIE(InfoExtractor):
         'only_matching': True,
     }]
 
+    _ACCESS_TOKEN = None
+    _REFRESH_TOKEN = None
+    _NETRC_MACHINE = 'fancode'
+
+    headers = {
+        'content-type': 'application/json',
+        'origin': 'https://fancode.com',
+        'referer': 'https://fancode.com',
+    }
+
+    def _login(self):
+        # Access tokens are shortlived, so get them using the refresh token.
+        username, password = self._get_login_info()
+        if username == 'refresh' and password is not None:
+            self.report_login()
+            data = '''{
+                "query":"mutation RefreshToken($refreshToken: String\\u0021) { refreshToken(refreshToken: $refreshToken) { accessToken }}",
+                "variables":{
+                    "refreshToken":"%s"
+                },
+                "operationName":"RefreshToken"
+            }''' % password
+
+            token_json = self.download_gql('refresh token', data, "Getting the Access token")
+            self._ACCESS_TOKEN = try_get(token_json, lambda x: x['data']['refreshToken']['accessToken'])
+            if self._ACCESS_TOKEN is None:
+                raise ExtractorError('Failed to get Access token')
+            else:
+                self._REFRESH_TOKEN = password
+                self.headers.update({'Authorization': 'Bearer %s' % self._ACCESS_TOKEN})
+        else:
+            pass
```

Comment:
If some other username was given, warn the user about how to actually use the option